apm-agent-core/src/main/java/co/elastic/apm/agent/bci/AgentMain.java

```diff
@@ -90,6 +98,43 @@ public synchronized static void init(String agentArguments, Instrumentation inst
         }
     }
 
+    /**
+     * Checks if a given version of the JVM is supported by this agent.
+     * Supports values provided before and after https://openjdk.java.net/jeps/223
+     *
+     * @param javaVersion jvm version, from {@code System.getProperty("java.version")}
+     * @return true if the version is supported, false otherwise
+     */
+    // package-protected for testing
+    static boolean isJavaVersionSupported(String javaVersion) {
+        boolean postJsr223 = !javaVersion.startsWith("1.");
+        // new scheme introduced in java 9, thus we can use it as a shortcut
+        if (postJsr223) {
+            return true;
+        }
+
+        char major = javaVersion.charAt(2);
+        if (major < '7') {
+            // given code is compiled with java 7, this one is unlikely in practice
+            return false;
+        } else if (major == '7' || major > '8') {
+            return true;
+        } else {
+            // java 8
+            int updateIndex = javaVersion.lastIndexOf("_");
+            if (updateIndex <= 0) {
+                return false;
+            } else {
+                String updateVersion = javaVersion.substring(updateIndex + 1);
```

Comment:
One of our Tomcat tests fails due to the version called: `1.8.0_72-internal` 🙁

Reply:
fixed in the last commit.